### PR TITLE
Feature Request: Search box above section list #277

### PR DESCRIPTION
**Pull request by ksaswin**
Add feature requested in issue #277

Add new SectionFilter component to filter slugs based on section names
Update SectionsColumn to display filtered sections
Add units tests for SectionFilter component

A screenshot of the implementation:
image

### CODE REVIEW

**Comment by octokatherine** [comment]
Hey @ksaswin, thanks for this contribution, I love the concept! 

A couple small things:

- can we move the search box above the "Custom section" button?
- when a section is clicked on, can we clear out the search input? I think the UX would feel a little nicer

**Comment by ksaswin** [comment]
Hi @octokatherine,

Yeah, sounds good to me. I'll start working on those changes right away 😊

**Comment by ksaswin** [comment]
Hi @octokatherine,

I've pushed the requested changes. Please feel free to let me know if you'd like me to make any more updates to the same.

Thank you 😊

File: components/SectionsColumn.js [hunk 1]
```diff
@@ -76,13 +79,19 @@ export const SectionsColumn = ({
...
   const onAddSection = (e, section) => {
     localStorage.setItem('current-focused-slug', section)
     setpageRefreshed(false)
     setAddAction(true)
-    setSectionSlugs((prev) => prev.filter((s) => s !== section))
+    setSectionSlugs((prev) => updateSlugsOnAdd(prev, section))
+    setFilteredSlugs((prev) => updateSlugsOnAdd(prev, section))
     setSelectedSectionSlugs((prev) => [...prev, section])
     setFocusedSectionSlug(localStorage.getItem('current-focused-slug'))
+    resetSearchFilter('')
```
Comment (octokatherine) [inline components/SectionsColumn.js:94]:
```suggestion
    resetSearchFilter()
```

don't think you need that argument!

**Review by octokatherine — COMMENTED** [review]
small suggestion, but it's looking good!

File: components/SectionsColumn.js (hunk 1 above)
Comment (ksaswin) [reply components/SectionsColumn.js]: Hi @octokatherine,

Nice catch 😄  I don't know how I missed this! I've pushed the requested changes.

Thank you 😊

**Pull request merged**